<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SetupWindow</class>
 <widget class="QMainWindow" name="SetupWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>276</width>
    <height>306</height>
   </rect>
  </property>
  <property name="contextMenuPolicy">
   <enum>Qt::NoContextMenu</enum>
  </property>
  <property name="windowTitle">
   <string>Crossbar Setup</string>
  </property>
  <property name="animated">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="1" column="0">
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="spacing">
       <number>6</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="spacing">
           <number>6</number>
          </property>
          <property name="sizeConstraint">
           <enum>QLayout::SetDefaultConstraint</enum>
          </property>
          <item>
           <widget class="QLineEdit" name="sizeEdit">
            <property name="placeholderText">
             <string>Size of the crossbar</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="dataQubitsEdit">
            <property name="placeholderText">
             <string>Number of data qubits</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="ancillaQubitsEdit">
            <property name="placeholderText">
             <string>Number of ancilla qubits</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="createButton">
            <property name="text">
             <string>Create </string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line">
            <property name="lineWidth">
             <number>1</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="loadJSON">
            <property name="text">
             <string>Load JSON topology</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="text">
    <string>Settings</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open...</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
 <slots>
  <slot>change_dir()</slot>
  <slot>validate_dir()</slot>
  <slot>filter_state_changed()</slot>
  <slot>validate_regex()</slot>
 </slots>
</ui>
